src/classes/EscalationTeamTrigger.cls

```diff
@@ -0,0 +1,140 @@
+		get {
+			return (Map<Id, rh_escal__EscalationTeam__c>) ((BaseTrigger) this).baseOldMap;
+		}
+		set;
+	}
+
+	/** The new map of objects */
+	@TestVisible Map<Id, rh_escal__EscalationTeam__c> newMap {
+		get {
+			if (this.newMap == null) {
+				this.newMap = new Map<Id, rh_escal__EscalationTeam__c>();
+
+				if (!this.isBefore && !this.newObjs.isEmpty()) {
+					this.newMap = new Map<Id, rh_escal__EscalationTeam__c>(this.newObjs);
+				}
+			}
+
+			return this.newMap;
+		}
+		set;
+	}
+
+	/** The new list of objects */
+	@TestVisible List<rh_escal__EscalationTeam__c> newObjs {
+		get {
+			return (List<rh_escal__EscalationTeam__c>) ((BaseTrigger) this).baseNewObjs;
+		}
+		set;
+	}
+
+	/**
+	* The constructor
+	*
+	* @param xoldMap The old map from the trigger
+	* @param xnewObj The list of new objects from the trigger
+	* @param isBefore If the trigger is in before or after
+	*/
+	public EscalationTeamTrigger(Map<Id, rh_escal__EscalationTeam__c> xoldMap, List<rh_escal__EscalationTeam__c> xnewObjs, Boolean isBefore) {
+		super(xoldMap, xnewObjs, isBefore);
+	}
+
+	/**
+	* Sets the name of the trigger
+	*
+	* @return The name of the trigger
+	*/
+	public override String getTriggerName() {
+		return 'EscalationTeamTrigger';
+	}
+
+	/**
+	* Do not allow addition of duplicate team members on an escalation
+	*/
+	@TestVisible
+	private void teamDupCheck() {
+		if (this.isDelete) {
+			return;
+		}
+
+		Set<Id> escalationIds = new Set<Id>();
+
+		for (rh_escal__EscalationTeam__c team: this.newObjs) {
+			escalationIds.add(team.rh_escal__Escalation__c);
+		}
+
+		escalationIds.remove(null);
+
+		if (escalationIds.isEmpty()) {
+			return;
+		}
+
+		Map<Id, Set<Id>> escalToMembers = new Map<Id, Set<Id>>();
+		Map<Id, Set<String>> escalToEmails = new Map<Id, Set<String>>();
+
+		for (rh_escal__EscalationTeam__c team: EscalationTeamUtils.getEscalationTeamMembers(escalationIds)) {
+			if (!escalToMembers.containsKey(team.rh_escal__Escalation__c)) {
+				escalToMembers.put(team.rh_escal__Escalation__c, new Set<Id>());
+				escalToEmails.put(team.rh_escal__Escalation__c, new Set<String>());
+			}
+
+			if (team.rh_escal__Contact__c != null) {
+				escalToMembers.get(team.rh_escal__Escalation__c).add(team.rh_escal__Contact__c);
+			} else if (team.rh_escal__User__c != null) {
+				escalToMembers.get(team.rh_escal__Escalation__c).add(team.rh_escal__User__c);
+			} else if (team.rh_escal__Email__c != null) {
+				escalToEmails.get(team.rh_escal__Escalation__c).add(team.rh_escal__Email__c);
+			}
+		}
+
+		for (rh_escal__EscalationTeam__c team: this.newObjs) {
+			Set<Id> members = escalToMembers.get(team.rh_escal__Escalation__c);
+			Set<String> emails = escalToEmails.get(team.rh_escal__Escalation__c);
+
+			Boolean isContactNull = team.rh_escal__Contact__c == null;
+			Boolean isUserNull = team.rh_escal__User__c == null;
+			Boolean isEmailNull = team.rh_escal__Email__c == null;
+			Boolean isMembersNull = members == null;
```

Comment:
With the update above, we should be able to get rid of `isMembersNull` and `isEmailsNull` here and in the checks on lines 111-115